### PR TITLE
[ONME-2731] Support for mbedOS-5

### PR DESCRIPTION
**Pull request by hasnainvirk**
* yotta_config.h is renamed to mbed_config.h
* A section is added in mbed_config.h to interprate mbedOS-5 config.
* A memory leak is fixed in the call of mac_api (in case of plugging-in and plugging out)
  of ethernet cable. The idea is to create only one instance of mac_api and reuse it after the
  cable is plugged back in again.
* Some traces are changed to info level. The whole idea is to reduce no. of traces caught at serial
  output. Too many traces reduce the performance by a fair amount.

### CODE REVIEW

**Comment by hasnainvirk** [comment]
@SeppoTakalo @TuomoHautamaki

File: source/borderrouter_tasklet.c [hunk 1]
```diff
@@ -333,7 +332,9 @@ static int8_t rf_interface_init(void)
         storage_sizes.key_description_table_size = 3;
         storage_sizes.key_lookup_size = 1;
         storage_sizes.key_usage_size = 3;
-        mac_api_t *api = ns_sw_mac_create(rf_phy_device_register_id, &storage_sizes);
+        if(!api) {
+        api = ns_sw_mac_create(rf_phy_device_register_id, &storage_sizes);
```
Comment (SeppoTakalo) [inline source/borderrouter_tasklet.c]: Indent

File: source/borderrouter_tasklet.c [hunk 2]
```diff
@@ -59,6 +55,9 @@ conf_t *global_config = static_config;
 #define NR_BACKHAUL_INTERFACE_PHY_DRIVER_READY 2
 #define NR_BACKHAUL_INTERFACE_PHY_DOWN  3
 
+static mac_api_t *api;
```
Comment (SeppoTakalo) [inline source/borderrouter_tasklet.c]: Later on, you assume *api to default to `NULL`
Please initialize this to `NULL` just to be safe.

Comment (SeppoTakalo) [inline source/borderrouter_tasklet.c]: Invalid comment. Kevin told that by default they are initialized to 0.
Ignore my comment...

File: source/borderrouter_tasklet.c [hunk 3]
```diff
@@ -368,8 +369,11 @@ static int backhaul_interface_up(int8_t driver_id)
     if (backhaul_if_id != -1) {
         tr_debug("Border RouterInterface already at active state\n");
     } else {
-        eth_mac_api_t *api = ethernet_mac_create(driver_id);
-        backhaul_if_id = arm_nwk_interface_ethernet_init(api, "bh0");
+        if(!eth_mac_api) {
+            eth_mac_api = ethernet_mac_create(driver_id);
+            }
```
Comment (SeppoTakalo) [inline source/borderrouter_tasklet.c]: Weird indentation.

**Comment by SeppoTakalo** [comment]
Looks good. Fix those couple of indentations.

**Pull request merged**